src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -2401,6 +2407,29 @@ private bool StartSourceChecksumCalculation(PEModuleBuilder moduleBeingBuilt, Di
                 return false;
             }
 
+            // Add debug documents for all embedded text first. This ensures that embedding takes priority over the 
```

Comment:
I'm going to revert b38f5d39af261177d248e726c6855ff39ced2a5f (line directive map refactoring) as part of the fix for this. The map is insufficient to get a good location for the diagnostic. I will instead defer this part to language specific override. Some refactoring of line directive map to share more code and share empty instances might still be beneficial, but would be completely unrelated to this PR, so I'll file an issue for that.